Utils/NamedTaskQueue.swift

```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Deferred
+
+private let DefaultDispatchQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)
+
+// NamedTaskQueue is a mutable task queue, terminated by a `Deferred<Maybe<[String:T]>>`
+// The initiator task takes no arguments, and returns a Deferred<Maybe<S>>.
+// Once this has run, an appendable list of named tasks are run, of signature
+// `(S) -> Deferred<Maybe<T>>`.
+// You add named tasks, which are executed on a dispatch queue.
+// Once the tasks are all executed, then terminal is filled with the results from all tasks that were added.
+// If any task, including the initiator fails, then the terminal is filled with a failure Maybe.
+// This differs from `DeferredUtils.all` in that:
+//  * tasks can be added after construction.
+//  * only one task for a given name is executed.
+// Tasks are `(S) -> Deferred<Maybe<T>>`, added with the `append:name:task` method.
+// The result of running all the tasks are accessed either as a `Deferred<Maybe<[String:Maybe<T>]>>`, or by
+// adding an `upon` block to the queue.
+// If the queue has finished running all tasks, then no more tasks can be added, and a new queue is needed.
+public class NamedTaskQueue<S, T> {
+
+    public typealias Task = (S) -> Deferred<Maybe<T>>
+
+    public typealias NamedTask = (String, Task)
+
+    public typealias AccumulatedTaskResults = [String: Maybe<T>]
```

Comment:
This is the `U` in a typical reducer. In a typical reducer it's a sequence, not a map. Note that if you give tasks the responsibility for how to reduce, you don't need them to expose their name. (You can do this in a wrapper.)